includes/Core/Conversion_Tracking/Conversion_Event_Providers/WooCommerce.php

```diff
@@ -66,21 +66,24 @@ public function register_script() {
 
 		$script->register( $this->context );
 
-		$this->add_purchase_event_hook();
-
 		return $script;
 	}
 
 	/**
 	 * Adds a hook for a purchase event.
+	 *
+	 * @since 1.129.0
 	 */
-	public function add_purchase_event_hook() {
+	public function register_hooks() {
+		$is_active = $this->is_active();
+		$input     = $this->context->input();
+
 		add_action(
 			'woocommerce_thankyou',
-			function ( $order_id ) {
+			function ( $order_id ) use ( $input, $is_active ) {
 				// Don't output this script tag if conversion tracking is
 				// disabled.
-				if ( ! $this->is_active() ) {
+				if ( ! $is_active ) {
 					return;
 				}
```

Comment:
This check is not needed, because the `register_scripts`  hook will only be called for the `$active_providers` returned by  `Conversion_Tracking::get_active_providers` which already ensures the provider is active by calling the `is_active` function among other things.